### PR TITLE
Allow the ballerina central access token to be specified as an environment variable when pushing modules

### PR DESCRIPTION
**Pull request by NatashaWso2**
Following environment variable should be specified with the access token.
```bash
BALLERINA_CENTRAL_ACCESS_TOKEN="xxxx-xxxx-xxxx-xxxx"
```

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/14241

## Automation tests
 - Integration tests
   > Added

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/PushUtils.java [hunk 1]
```diff
@@ -139,8 +141,24 @@ public static void pushPackages(String packageName, String sourceRoot, String in
...
         
         if (installToRepo == null) {
-            // Get access token
-            String accessToken = checkAccessToken();
+            // The access token can be specified as an environment variable or in 'Settings.toml'. First we would
+            // check if the access token was specified as an environment variable. If not we would read it from
+            // 'Settings.toml'.
+
+            // Check if the access token is specified as an environment variable
+            String accessToken = System.getenv("BALLERINA_CENTRAL_ACCESS_TOKEN");
```
Comment (lafernando) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/PushUtils.java]: It's better if can put "BALLERINA_CENTRAL_ACCESS_TOKEN" to a constant first.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/PushUtils.java [hunk 2]
```diff
@@ -139,8 +141,24 @@ public static void pushPackages(String packageName, String sourceRoot, String in
...
+            if (accessToken != null) {
```
Comment (lafernando) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/PushUtils.java]: I think we can cleanup this code design a bit. Rather than, doing this logic here, and creating redundant code, update the code in "TomlParserUtils.readSettings()" method. Here, simply check if a Settings.toml is there in the home directory, if not, create empty values for those, Central object etc... and also check for the existence of the environment variable here for "accessToken", or else, keep it the original value (coming from Settings.toml, if not there, empty). Then, the rest of the existing logic can be kept in-tact, and no code duplication is required.

File: tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/ModulePushTestCase.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                projectPath.toString());
+
+        // Push the module to Ballerina Central
+        String msg = orgName + "/" + moduleName + ":0.0.1 [project repo -> central]";
+        LogLeecher logLeecher = new LogLeecher(msg);
+        balClient.runMain("push", new String[]{moduleName}, envVariables, new String[]{},
+                new LogLeecher[]{logLeecher}, projectPath.toString());
+        logLeecher.waitForText(5000);
+    }
+
```
Comment (Shan1024) [inline tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/ModulePushTestCase.java]: Extra newline.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/PushUtils.java (hunk 1 above)
Comment (NatashaWso2) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/PushUtils.java]: Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/14242/commits/051de5130e23039496be15fe8469f93bb3505e10

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/PushUtils.java (hunk 2 above)
Comment (NatashaWso2) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/PushUtils.java]: Refactored the code with https://github.com/ballerina-platform/ballerina-lang/pull/14242/commits/051de5130e23039496be15fe8469f93bb3505e10

File: tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/ModulePushTestCase.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        moduleName = moduleName + PackagingTestUtils.randomModuleName(8);
+        envVariables = addEnvVariables(PackagingTestUtils.getEnvVariables());
+    }
+
+    @Test(description = "Test pushing a package to central by specifying the access token as an environment variable")
+    public void testPush() throws Exception {
+        // Org-name
+        String orgName = "integrationtests";
+
+        // Initialize a ballerina project and create a module to be pushed
```
Comment (Shan1024) [inline tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/ModulePushTestCase.java]:
```suggestion
        // Initialize a ballerina project and create a module to be pushed.
```

File: tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/ModulePushTestCase.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        Path projectPath = tempProjectDirectory.resolve("initPushProject");
+        Files.createDirectories(projectPath);
+        String[] options = {"\n", orgName + "\n", "\n", "m\n", moduleName + "\n", "f\n"};
+        balClient.runMain("init", new String[]{"-i"}, envVariables, options, new LogLeecher[]{},
+                projectPath.toString());
+
+        // Push the module to Ballerina Central
```
Comment (Shan1024) [inline tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/ModulePushTestCase.java]:
```suggestion
        // Push the module to Ballerina Central.
```

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/PushUtils.java [hunk 6]
```diff
@@ -63,14 +63,15 @@
 public class PushUtils {
 
     private static final String BALLERINA_CENTRAL_CLI_TOKEN = "https://central.ballerina.io/cli-token";
-    private static final PrintStream SYS_ERR = System.err;
     private static final Path BALLERINA_HOME_PATH = RepoUtils.createAndGetHomeReposPath();
     private static final Path SETTINGS_TOML_FILE_PATH = BALLERINA_HOME_PATH.resolve(
-            ProjectDirConstants.SETTINGS_FILE_NAME);
-    private static PrintStream outStream = System.out;
+                                                                        ProjectDirConstants.SETTINGS_FILE_NAME);
```
Comment (Shan1024) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/PushUtils.java]: Better to revert this change? Earlier formatting looks better. WDYT?

File: tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/ModulePushTestCase.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+
+    @Test(description = "Test pushing a package to central by specifying the access token as an environment variable")
+    public void testPush() throws Exception {
+        // Org-name
+        String orgName = "integrationtests";
+
+        // Initialize a ballerina project and create a module to be pushed
+        Path projectPath = tempProjectDirectory.resolve("initPushProject");
+        Files.createDirectories(projectPath);
+        String[] options = {"\n", orgName + "\n", "\n", "m\n", moduleName + "\n", "f\n"};
```
Comment (lafernando) [inline tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/ModulePushTestCase.java]: Minor comments, but this is not how you should properly format the code, the indentation is incorrect, it should be { "\n" ... there should be a space after "{". 

Also, below "new String[]{" should be "new String[] { moduleName }", new LogLeecher[]{} should be new LogLeeher[] {} .. please fix in all similar places.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/PushUtils.java (hunk 6 above)
Comment (lafernando) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/PushUtils.java]: +1.

File: tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/ModulePushTestCase.java (hunk 7 above)
Comment (Shan1024) [reply tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/ModulePushTestCase.java]: I think this is the current formatting which we have used throughout the code base. So I think it is better to stick to the same format here as well. WDYT?

Comment (lafernando) [reply tests/ballerina-tools-integration-test/src/test/java/org/ballerinalang/test/packaging/ModulePushTestCase.java]: hmm.. I see, then it's fine for now.

**Comment by rukmal** [comment]
This does not appear to be documented anywhere on the website, and is only mentioned in the [Ballerina Action](https://github.com/ballerina-platform/ballerina-action/blob/master/README.md) README.

This should be added to https://ballerina.io/learn/cli-documentation/cli-commands/.
